src/sentry/integrations/slack/notifications.py

```diff
@@ -32,26 +32,30 @@ def get_context(
 def get_channel_and_integration_by_user(
     user: User, organization: Organization
 ) -> Tuple[Optional[str], Optional[Integration]]:
-    try:
-        identity = Identity.objects.get(
-            idp__type=EXTERNAL_PROVIDERS[ExternalProviders.SLACK],
-            user=user.id,
-        )
-    except Identity.DoesNotExist:
+
+    identities = Identity.objects.filter(
+        idp__type=EXTERNAL_PROVIDERS[ExternalProviders.SLACK],
+        user=user.id,
+    )
```

Comment:
Since you're accessing the `idp` later on, probably worth using `.select_related('idp')`

Reply:
This is still worth doing